### PR TITLE
Introduce PowerPC architecture to qiling

### PR DESCRIPTION
**Pull request by bet4it**
### Which kind of PR do you create?

- [ ] This PR only contains minor fixes.
- [x] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [x] Essential comments are added.
- [x] The reference of the new code is pointed out.

### Extra tests?

- [ ] No extra tests are needed for this PR.
- [x] I have added enough tests for this PR.
- [ ] Tests will be added after some discussion and review.

### Changelog?

- [ ] This PR doesn't need to update Changelog.
- [x] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

The test binaries are provided in qilingframework/rootfs#29

### CODE REVIEW

**Comment by xwings** [comment]
There is still some error in CI after adding the file.

File: qiling/arch/ppc_const.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    "f25": UC_PPC_REG_FPR25,
+    "f26": UC_PPC_REG_FPR26,
+    "f27": UC_PPC_REG_FPR27,
+    "f28": UC_PPC_REG_FPR28,
+    "f29": UC_PPC_REG_FPR29,
+    "f30": UC_PPC_REG_FPR30,
+    "f31": UC_PPC_REG_FPR31,
+}
+
+class MSR(IntEnum):
```
Comment (elicn) [inline qiling/arch/ppc_const.py:85]: I think that `Flag` ot `IntFlag` would be a better fit here (see [here](https://docs.python.org/3.8/library/enum.html#enum.Flag)).

**Review by elicn — APPROVED** [review]
Great contribution!
I am curious whether extending Qiling with a new architecture was an easy task, or not..?

File: qiling/arch/ppc_const.py (hunk 1 above)
Comment (bet4it) [reply qiling/arch/ppc_const.py:85]: You could search `IntEnum` in the code. There are many usages of `IntEnum`, but none of `Flag` or `IntFlag`. So if we really want to change it, that could be another PR.

**Review by xwings — APPROVED** [review]

**Comment by xwings** [comment]
Amazing!

**Comment by xwings** [comment]
Thanks !

**Pull request merged**